### DISCUSSION TITLE
Parser Exception

### DISCUSSION THREAD

**Discussion by mshearer0** [moj-analytical-services/splink#889] [General]
I'm getting the following parser exception when trying to perform a simple match:

```
ParserException: Parser Error: syntax error at or near ")"
LINE 11: ... CREATE TABLE __splink__m_u_counts_827346a
        AS
        (WITH __splink__df_comparison_vectors as (select * from __splink__df_comparison_vectors_cde9c45), 
__splink__df_match_weight_parts as (
    select "source_dataset_l","unique_id_l","source_dataset_r","unique_id_r","year_l","year_r","month_l","month_r",match_key 
    from __splink__df_comparison_vectors
    ), 
__splink__df_predict as (
    select
    log2(cast(0.10241024102410243 as double) * ) as match_weight,
```

Any suggestions welcome.

**Comment by ADBond** [comment]
Do you have any details on the code/linker `settings` that led to this error?

One way this can happen is if you try to call `linker.predict()` when you don't have any `comparisons` (this should be updated so the error is clearer!), in which case there is nothing for the model to do, as every record pair will have a match probability of `probabillity_two_random_records_match`.

**Reply by mshearer0** [reply]
Thanks, that could be it. The number of potential matches is small. I started with a larger dataset but cut it down to run on laptop and could have removed all the matches in the process. Will revert to the larger set with definite matches once I figure out how to get spark running.
